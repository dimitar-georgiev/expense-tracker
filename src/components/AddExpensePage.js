/**
 * Created by Mitaka on 17-Nov-17.
 */
import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {addExpense} from '../actions/expenses';

const  AddExpensePage = (props) => (
    <div>
        <h2>Add Expense</h2>
        <ExpenseForm
            onSubmit={(expense) => {
                //console.log("Expense: ",expense);
                props.dispatch(addExpense(expense));
                props.history.push('/');
            }}
        />
    </div>
);

export default connect()(AddExpensePage);