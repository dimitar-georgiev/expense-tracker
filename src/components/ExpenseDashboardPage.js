/**
 * Created by Mitaka on 17-Nov-17.
 */
import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const  ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
);

export default ExpenseDashboardPage;