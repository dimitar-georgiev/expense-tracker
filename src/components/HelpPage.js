/**
 * Created by Mitaka on 17-Nov-17.
 */
import React from 'react';

const  HelpPage = () => (
    <div>
        Help Page
    </div>
);

export default HelpPage;