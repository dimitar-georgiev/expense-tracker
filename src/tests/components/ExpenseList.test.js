/**
 * Created by Mitaka on 26-Nov-17.
 */
import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('render ExpensesList', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />);

    expect(wrapper).toMatchSnapshot();
});

test('print No expenses when no expenses provided', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);

    expect(wrapper).toMatchSnapshot();
});