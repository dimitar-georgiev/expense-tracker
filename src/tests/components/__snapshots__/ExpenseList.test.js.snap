// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print No expenses when no expenses provided 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;

exports[`render ExpensesList 1`] = `
<div>
  <ExpenseListItem
    amount={0}
    createdAt={0}
    description="Test Description Edno"
    id={1}
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={787}
    createdAt={-345600000}
    description="Test Description Dve"
    id={2}
    key="2"
    note="Test 2 Note"
  />
  <ExpenseListItem
    amount={147}
    createdAt={345600000}
    description="Test Description Tri"
    id={3}
    key="3"
    note=""
  />
</div>
`;
